rest/server_context.go

```diff
@@ -1065,6 +1065,27 @@ func (sc *ServerContext) updateCalculatedStats() {
 
 }
 
+// Record Goroutines high watermark into expvars
+func recordGoroutineHighwaterMark(stats *expvar.Map, numGoroutines uint64) (maxGoroutinesSeen uint64) {
+
+	maxGoroutinesSeen = atomic.LoadUint64(&base.MaxGoroutinesSeen)
+
+	if numGoroutines > maxGoroutinesSeen {
+
+		// Ignore CAS failures -- this stat can be considered a "best effort"
+		_ = atomic.CompareAndSwapUint64(&base.MaxGoroutinesSeen, maxGoroutinesSeen, numGoroutines)
```

Comment:
If we're ignoring CompareAndSwap errors, we could probably just stick with atomic.StoreUint64 for simplicity.  (it's 50/50 that we get the right result whether we do last write wins or first write wins in a concurrent update scenario).

Reply:
I agree this is less confusing.  Will update.